### PR TITLE
[Messenger] Display a nice error when connection fail

### PR DESCRIPTION
**Pull request by lyrixx**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | kind of
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets |
| License       | MIT
| Doc PR        | 

### Before:

image

### Now:

image

### CODE REVIEW

**Review by weaverryan — APPROVED** [review]

File: src/Symfony/Component/Messenger/Transport/AmqpExt/Connection.php [hunk 1]
```diff
@@ -201,10 +201,11 @@ public function channel(): \AMQPChannel
             $connection = $this->amqpFactory->createConnection($this->connectionCredentials);
             $connectMethod = 'true' === ($this->connectionCredentials['persistent'] ?? 'false') ? 'pconnect' : 'connect';
 
-            if (false === $connection->{$connectMethod}()) {
-                throw new \AMQPException('Could not connect to the AMQP server. Please verify the provided DSN.');
+            try {
+                $connection->{$connectMethod}();
+            } catch (\AMQPConnectionException $e) {
+                throw new \AMQPException(sprintf('Could not connect to the AMQP server. Please verify the provided DSN. (%s)', json_encode($this->connectionCredentials)), 0, $e);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Messenger/Transport/AmqpExt/Connection.php]: what about turning the password into `*****`?

Comment (lyrixx) [reply src/Symfony/Component/Messenger/Transport/AmqpExt/Connection.php]: IIRC, the password is already in the trace

Comment (weaverryan) [reply src/Symfony/Component/Messenger/Transport/AmqpExt/Connection.php]: Even so, I think only this top message would make it into the logs - seems like a good idea.

Comment (lyrixx) [reply src/Symfony/Component/Messenger/Transport/AmqpExt/Connection.php]: Good catch. I will do that

Comment (lyrixx) [reply src/Symfony/Component/Messenger/Transport/AmqpExt/Connection.php]: Fixed :+1:

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @lyrixx.

**Pull request merged**